ext/puma_http11/mini_ssl.c

```diff
@@ -409,7 +437,7 @@ VALUE engine_extract(VALUE self) {
   ms_conn* conn;
   int bytes;
   size_t pending;
-  char buf[512];
+  char buf[4096];
```

Comment:
Why did this change size?

Reply:
I started with just the context change, and was testing with small response bodies.  Nate noticed an issue that appeared with longer response bodies, I then started debugging `MiniSSL::Socket#write` and the various read/write operations in it.

I didn't note it in the code, but the OpenSSL 1.1.1 man pages show 4096 is the default, and I verified that length was the max length returned by `MiniSSL::Engine#extract`.

I didn't look at the corresponding operation on the read side, thought I'd leave it as is.

JFYI, I'm working on a test update, and it has an ssl test with 200 kB response bodies...

Reply:
Maybe just throw a comment above this line why it's this size. I should have done that with the 512 actually.

Reply:
Done.  Maybe I got a little AR, comments below, easy to find next time...
```
  // https://www.openssl.org/docs/manmaster/man3/BIO_f_buffer.html
  // crypto/bio/bf_buff.c DEFAULT_BUFFER_SIZE
```